webapp/channels/src/components/post_view/gm_conversion_message/gm_conversion_message.tsx

```diff
@@ -36,8 +36,9 @@ function GMConversionMessage(props: Props): JSX.Element {
         ),
     );
 
-    const convertedByUserUsername = userProfiles.find((user) => user.id === convertedByUserId)!.username;
-    const gmMembersUsernames = userProfiles.map((user) => renderUsername(user.username));
+    // we can memoize on profile length as the number of users in a GM always stay constant
+    const convertedByUserUsername = useMemo(() => userProfiles.find((user) => user.id === convertedByUserId)!.username, [userProfiles.length]);
+    const gmMembersUsernames = useMemo(() => userProfiles.map((user) => renderUsername(user.username)), [userProfiles.length]);
```

Comment:
Same here, changes in the usernames won't be translated into this if we memoize only by the length.

Reply:
Removed memo as discussed.